lib/Optimizer/Transforms/LiftArrayAlloc.cpp

```diff
@@ -321,6 +366,88 @@ class ComplexCreatePattern : public OpRewritePattern<complex::CreateOp> {
   }
 };
 
+// Fold arith.trunc ops if the argument is constant.
+class FloatTruncatePattern : public OpRewritePattern<arith::TruncFOp> {
```

Comment:
I'm surprised these arith ops are showing up at all. Where are they coming from? Why isn't cc::CastOp used? Or handled?

Reply:
I encountered them when running for 32 bit precision (ie quantinuum) - the IR had float constants of double precision and truncation operation on them before creating complex

Reply:
I haven't seen casts before, but I can add a pattern when I encounter them

Reply:
Will add those to canonicalizer, among other things, in a separate PR, as discussed. For now, I moved all the patterns not related to lifting arrays to a new pass, ConstPropComplex